### DISCUSSION TITLE
Slow upload speed when using Cryptomator

### DISCUSSION THREAD

**Discussion by Chippel** [abraunegg/onedrive#2037] [Q&A] [answered]
I'm really happy I found a solution to use OneDrive on Linux. Unfortunately the CLI doesn't support encryption so I have to use a tool called Cryptomator to be able to encrypt my files before they are being uploaded to the cloud. 

It seems like it takes a really long time to upload my files. Is there any option to speed up the upload process? It seems like, when using -m, Onedrive only uploads 1 file at the time which is pretty much meh.

**Comment by abraunegg** [comment, marked as answer]
@Chippel 
> Unfortunately the CLI doesn't support encryption so I have to use a tool called Cryptomator to be able to encrypt my files before they are being uploaded to the cloud.

Its not just that the tool doesn't support encryption - OneDrive and the entire ecosystem behind it, is not geared to supporting encrypted uploaded content. If you want to store encrypted content on OneDrive, strongly consider that it is not fit-for-purpose.

Solutions that do support encrypted data uploads are:

* Amazon S3
* Backblaze B2
* Google Cloud Storage
* Google Drive

For further details please read: https://github.com/abraunegg/onedrive/issues/1023

> It seems like it takes a really long time to upload my files. Is there any option to speed up the upload process? It seems like, when using -m, Onedrive only uploads 1 file at the time which is pretty much meh.

Unfortunately this is how the client currently operates. Your experience is impacted by the following factors:
* Your Internet access speed
* The speed of your CPU
* How the client utilises the Curl library for each operation
* No multi-threaded capability at present - see: https://github.com/abraunegg/onedrive/issues/232
